test/testdrive/linked-clusters.td

```diff
@@ -0,0 +1,220 @@
+> DROP SINK snk
+
+# Test that creating a sink with the REMOTE option creates a linked cluster
+# with no replica. (Unsafe mode only.)
+> CREATE SINK snk FROM v
+  INTO KAFKA CONNECTION kafka_conn (TOPIC 'snk-${testdrive.seed}')
+  FORMAT JSON ENVELOPE DEBEZIUM WITH (REMOTE 'localhost:32323')
+> SHOW CLUSTERS
+default
+mz_introspection
+mz_system
+materialize_public_snk
+> SELECT cluster, replica, size FROM (SHOW CLUSTER REPLICAS)
+default                r1      ${arg.default-replica-size}
+mz_introspection       r1      1
+mz_system              r1      1
+> DROP SINK snk
+
+# Test that creating sinks with an explicit size creates linked clusters, each
+# with a replica named "linked" with the size specified in the sink.
+> CREATE SINK snk1 FROM v
+  INTO KAFKA CONNECTION kafka_conn (TOPIC 'snk1-${testdrive.seed}')
+  FORMAT JSON ENVELOPE DEBEZIUM WITH (SIZE = '1')
+> SHOW CLUSTERS
+default
+mz_introspection
+mz_system
+materialize_public_snk1
+> SELECT cluster, replica, size FROM (SHOW CLUSTER REPLICAS)
+default                  r1      ${arg.default-replica-size}
+mz_introspection         r1      1
+mz_system                r1      1
+materialize_public_snk1  linked  1
+> CREATE SINK snk2 FROM v
+  INTO KAFKA CONNECTION kafka_conn (TOPIC 'snk-${testdrive.seed}')
+  FORMAT JSON ENVELOPE DEBEZIUM WITH (SIZE = '2')
+> SHOW CLUSTERS
+default
+mz_introspection
+mz_system
+materialize_public_snk1
+materialize_public_snk2
+> SELECT cluster, replica, size FROM (SHOW CLUSTER REPLICAS)
+default                  r1      ${arg.default-replica-size}
+mz_introspection         r1      1
+mz_system                r1      1
+materialize_public_snk1  linked  1
+materialize_public_snk2  linked  2
+
+# Test that changing a sink size results in the cluster being dropped and
+# recreated.
+> ALTER SINK snk2 SET (SIZE = '4')
+> SELECT cluster, replica, size FROM (SHOW CLUSTER REPLICAS) WHERE cluster = 'materialize_public_snk2'
+materialize_public_snk2  linked  4
+
+# Test that a sink can be converted to a REMOTE sink and then back to a managed
+# sink. (Unsafe mode only.)
+> ALTER SINK snk2 SET (REMOTE 'localhost:32323')
+> SELECT cluster, replica, size FROM (SHOW CLUSTER REPLICAS) WHERE cluster = 'materialize_public_snk2'
+> ALTER SINK snk2 SET (SIZE = '4')
+> SELECT cluster, replica, size FROM (SHOW CLUSTER REPLICAS) WHERE cluster = 'materialize_public_snk2'
+materialize_public_snk2  linked  4
+> ALTER SINK snk2 RESET (SIZE)
+> SELECT cluster, replica, size FROM (SHOW CLUSTER REPLICAS) WHERE cluster = 'materialize_public_snk2'
+materialize_public_snk2  linked  ${arg.default-storage-size}
+
+# Test that only the default clusters and replicas remain after dropping
+# the sinks.
+> DROP SINK snk1
+> DROP SINK snk2
+> SHOW CLUSTERS
+default
+mz_introspection
+mz_system
+> SELECT cluster, replica, size FROM (SHOW CLUSTER REPLICAS)
+default                 r1      ${arg.default-replica-size}
+mz_introspection        r1      1
+mz_system               r1      1
```

Comment:
We should also have a test that ensures that attempting to run statements on a linked cluster is handled gracefully:

```
SET cluster = materialize_public_lg;
SELECT * FROM mz_indexes;
```

Currently, this makes environmentd crash with: 

```
thread 'coordinator' panicked at 'called `Result::unwrap()` on an `Err` value: InstanceMissing(User(2))', materialize/src/adapter/src/coord/indexes.rs:38:69
```

Reply:
Omg, I can't believe I forgot about actually disabling queries on the cluster. 🙈 I'll fix.